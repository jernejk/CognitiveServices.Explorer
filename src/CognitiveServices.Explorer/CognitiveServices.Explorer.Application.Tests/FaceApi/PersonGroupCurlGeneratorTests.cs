using CognitiveServices.Explorer.Application.FaceApi;
using FluentAssertions;
using Snapshooter.Xunit;
using Xunit;

namespace CognitiveServices.Explorer.Application.Tests.FaceApi
{
    public class PersonGroupCurlGeneratorTests
    {
        [Fact]
        public void ShouldGenerateCreate()
        {
            new PersonGroupRequestGenerator()
                .Create("default", "Test group")
                .Should()
                .MatchSnapshot();
        }

        [Fact]
        public void ShouldGenerateCreateWithUserData()
        {
            new PersonGroupRequestGenerator()
                .Create("default", "Test group 2", "imageUrl: image")
                .Should()
                .MatchSnapshot();
        }

        [Fact]
        public void ShouldGenerateUpdate()
        {
            new PersonGroupRequestGenerator()
                .Update("default", "Test group 2", "imageUrl: image")
                .Should()
                .MatchSnapshot();
        }

        [Fact]
        public void ShouldGenerateList()
        {
            new PersonGroupRequestGenerator()
                .List()
                .Should()
                .MatchSnapshot();
        }

        [Fact]
        public void ShouldGenerateDelete()
        {
            new PersonGroupRequestGenerator()
                .Delete("default")
                .Should()
                .MatchSnapshot();
        }

        [Fact]
        public void ShouldGenerateTrain()
        {
            new PersonGroupRequestGenerator()
                .Train("default")
                .Should()
                .MatchSnapshot();
        }

        [Fact]
        public void ShouldGenerateCheckTraining()
        {
            new PersonGroupRequestGenerator()
                .CheckTraining("default")
                .Should()
                .MatchSnapshot();
        }
    }
}
