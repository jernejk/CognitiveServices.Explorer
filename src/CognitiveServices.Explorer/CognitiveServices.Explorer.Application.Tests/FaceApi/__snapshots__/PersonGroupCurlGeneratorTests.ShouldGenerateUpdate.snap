{
  "RequestName": null,
  "Headers": {},
  "RelativePath": "face/v1.0/persongroups/default",
  "Queries": null,
  "HttpMethod": "PATCH",
  "Body": "{\"name\":\"Test group 2\",\"userData\":\"imageUrl: image\"}",
  "BinaryContent": null,
  "ContentType": "application/json",
  "ExpectedHtppCode": 200,
  "TokenHeaderName": "Ocp-Apim-Subscription-Key",
  "Cost": {
    "ServiceName": "FaceApi",
    "Cost": 1.0,
    "Unit": "tr"
  },
  "CognitiveServiceDoc": "https://westus.dev.cognitive.microsoft.com/docs/services/563879b61984550e40cbbe8d/operations/563879b61984550f3039524a"
}
